core/kazoo_attachments/src/kz_att_s3.erl

```diff
@@ -65,3 +61,22 @@ convert_kv({K, V})
 convert_kv({<<"etag">> = K, V}) ->
     {K, binary:replace(V, <<$">>, <<>>, ['global'])};
 convert_kv(KV) -> KV.
+
+-spec get_map_values(kz_data:connection()) -> {ne_binary(), ne_binary(), ne_binary(), api_binary(), ne_binary()}.
+get_map_values(#{'bucket' := Bucket
+                ,'key' := Key
+                ,'secret' := Secret
+                }=Map) ->
+    Path = maps:get('path', Map, 'undefined'),
+    Host = maps:get('host', Map,  <<"s3.amazonaws.com">>),
+    {Bucket, Key, Secret, Path, Host}.
+
+-spec get_file_path(api_binary(), ne_binary(), ne_binary(), ne_binary()) -> list().
+get_file_path('undefined', DbName, DocId, AName) ->
+    kz_util:to_list(list_to_binary([DbName, "/", DocId, "_", AName]));
+get_file_path(Path, DbName, DocId, AName) ->
+    kz_util:to_list(list_to_binary([Path, "/", DbName, "/", DocId, "_", AName])).
+
+-spec get_s3_values(ne_binary()) -> {ne_binary(), ne_binary(), ne_binary(), ne_binary(), api_binary()}.
+get_s3_values(S3) ->
+    binary_to_term(base64:decode(S3)).
```

Comment:
@tsromana you still to handle the case where there is no host encoded.
something like 
```
case binary_to_term(base64:decode(S3)) of
    {_Bucket, _Key, _Secret, _Path} = V1 -> V1;
    {_Bucket, _Key, _Secret, _Path, _Host} = V2 -> V2
end.
```